OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/ImportKeysProxyActivity.java

```diff
@@ -177,7 +184,8 @@ private void processScannedContent(Uri uri) {
     }
 
     public void returnResult(Intent data) {
-        if (returnResult) {
+        boolean isInternal = getIntent().getBooleanExtra(EXTRA_IS_INTERNAL, false);
```

Comment:
I think this can be replaced by a check if the activity has been started by ACTION_SCAN_WITH_RESULT or ACTION_SCAN_IMPORT, because only ACTION_QR_CODE_API is used from the outside.